src/Common/tests/XunitTraitsDiscoverers/Discoverers/PlatformSpecificDiscoverer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Collections.Generic;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Xunit.TraitDiscoverers
+{
+    /// <summary>
+    /// This class discovers all of the tests and test classes that have
+    /// applied the PlatformSpecific attribute
+    /// </summary>
+    public class PlatformSpecificDiscoverer : ITraitDiscoverer
+    {
+        /// <summary>
+        /// Gets the trait values from the Category attribute.
+        /// </summary>
+        /// <param name="traitAttribute">The trait attribute containing the trait values.</param>
+        /// <returns>The trait values.</returns>
+        public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+        {
+            object[] ctorArgs = (object[])traitAttribute.GetConstructorArguments().First();
+            bool nonWindowsTest = true, nonLinuxTest = true, nonMacTest = true;
+            foreach (var arg in ctorArgs)
+            {
+                PlatformID platform = (PlatformID)arg;
+                switch (platform)
+                {
+                    case PlatformID.Windows:
+                        nonLinuxTest &= true;
+                        nonMacTest &= true;
```

Comment:
These &s with true aren't necessary, right?  The variables all start as true, so they only way they can change is from true to false.